### DISCUSSION TITLE
Using useEventSource in multiple places

### DISCUSSION THREAD

**Discussion by lockr7** [sergiodxa/remix-utils#214] [Q&A] [answered]
If I am using `useEventSource` with the same url/options, does it open a new connection for each instance? or is the same connection reused across all components?

**Comment by sergiodxa** [comment, marked as answer]
It opens a new connection.

**Reply by lockr7** [reply]
Appreciate the quick response! Is there any implementation pattern to share a connection?

**Reply by sergiodxa** [reply]
You will have to use your own hook to do that. You will need to probably keep a global map of connections so you can just reuse them.

**Reply by sergiodxa** [reply]
A future major version of Remix Utils will include support for this.

**Comment by incarnateTheGreat** [comment]
Sorry to ask, but has a solution for this been implemented? If so, I don't believe I see a working example in the docs.

To be clear, I would be making a call using `useEventSource` from the parent page route and then summon the same data in the parent's child components similar to `useRouteLoaderData`. 

It looks like you're using `context` to summon the data and using the `url` as the key. Whenever I use `useEventSource` with just the key, it summons a different instance of the event source and doesn't have the context code.

Thanks!

@lockr7 were you able to successfully use the updated implementation?
